src/universal_transfer_operator/integrations/hooks/fivetran.py

```diff
@@ -0,0 +1,361 @@
+        endpoint = self.api_metadata_path_connectors + connector_id + "/" + metadata
+        resp = self._do_api_call(("GET", endpoint))
+        return resp["data"]
+
+    def get_destinations(self, group_id) -> Any:
+        """
+        Fetches destination information for the given group.
+
+        :param group_id: The Fivetran group ID, returned by a connector API call.
+        """
+        if group_id == "":
+            raise ValueError("No value specified for group_id")
+        endpoint = self.api_path_destinations + group_id
+        resp = self._do_api_call(("GET", endpoint))
+        return resp["data"]
+
+    def get_groups(self, group_id) -> Any:
+        """
+        Fetches destination information for the given group.
+
+        :param group_id: The Fivetran group ID, returned by a connector API call.
+        """
+        if group_id == "":
+            raise ValueError("No value specified for connector_id")
+        endpoint = self.api_path_groups + group_id
+        resp = self._do_api_call(("GET", endpoint))
+        return resp["data"]
+
+    def check_connector(self, connector_id):
+        """
+        Ensures connector configuration has been completed successfully and is in a functional state.
+
+        :param connector_id: Fivetran connector_id, found in connector settings page in the Fivetran user interface.
+        """
+        connector_details = self.get_connector(connector_id)
+        service_name = connector_details["service"]
+        schema_name = connector_details["schema"]
+        setup_state = connector_details["status"]["setup_state"]
+        if setup_state != "connected":
+            raise AirflowException(
+                f'Fivetran connector "{connector_id}" not correctly configured, '
+                f"status: {setup_state}\nPlease see: "
+                f"{self._connector_ui_url_setup(service_name, schema_name)}"
+            )
+        self.log.info(f"Connector type: {service_name}, connector schema: {schema_name}")
+        self.log.info(f"Connectors logs at {self._connector_ui_url_logs(service_name, schema_name)}")
+        return True
+
+    def set_schedule_type(self, connector_id, schedule_type):
+        """
+        Set connector sync mode to switch sync control between API and UI.
+
+        :param connector_id: Fivetran connector_id, found in connector settings page in the Fivetran user interface.
+        """
+        endpoint = self.api_path_connectors + connector_id
+        return self._do_api_call(("PATCH", endpoint), json.dumps({"schedule_type": schedule_type}))
+
+    def prep_connector(self, connector_id, schedule_type):
+        """
+        Prepare the connector to run in Airflow by checking that it exists and is a good state, then update connector
+        sync schedule type if changed.
+
+        :param connector_id: Fivetran connector_id, found in connector settings page in the Fivetran user interface.
```

Comment:
add schedule_type to params

Reply:
[Done](https://github.com/astronomer/apache-airflow-provider-transfers/pull/62/commits/9fd6e7f7850c27f22e10d44a17e5f5c0744a15f9)